src/cloudimagedirectory/filter/filter.py

```diff
@@ -15,15 +19,44 @@ def get_utc_datetime(date_string: str) -> pd.Timestamp:  # type: ignore[no-any-u
 def FilterImageByFilename(word: str) -> Callable:
     """Filter images by filename."""
     print("filter images by filename: " + word)
-    return lambda data: [d for d in data if not d.filename.lower().__contains__(word.lower())]
+
+    filtered_image_by_filename_counter = meter.create_counter(
+        name="filtered_image_by_filename_counter",
+        description="Counts the number of filtered images by filename",
+        unit="1",
+    )
+
+    def _filter_image_by_filename(data: str) -> list:
+        result = []
+        for d in data:
+            if not d.filename.lower().__contains__(word.lower()):
+                result.append(d)
+            else:
+                filtered_image_by_filename_counter.add(1, {"word", word})
+        return result
+    return _filter_image_by_filename
 
 
 def FilterImageByLatestUpdate(latestDate: pd.Timestamp) -> Callable:  # type: ignore[no-any-unimported]
     """Filter images by latest date."""
     print(f"filter images by latest date: {latestDate}")
     latestDate = latestDate.replace(tzinfo=pytz.UTC)
 
-    return lambda data: [d for d in data if d.content is not None and get_utc_datetime(d.content["date"]) > latestDate]
+    filtered_image_by_latest_update_counter = meter.create_counter(
+        name="filtered_image_by_latest_update_counter",
+        description="Counts the number of filtered images by latest update",
+        unit="1",
+    )
+
+    def _filter_image_by_latest_update(data: str) -> list:
+        result = []
+        for d in data:
+            if d.content is not None and get_utc_datetime(d.content["date"]) > latestDate:
+                result.append(d)
+            else:
+                filtered_image_by_latest_update_counter.add(1, {"latest_date", latestDate})
+        return result
+    return _filter_image_by_latest_update
```

Comment:
Just for info, another useful way of filtering items is to use the [filter](https://docs.python.org/3/library/functions.html#filter) function. For example:

```python

def _is_latest(d) -> bool:
    return d.content is not None and get_utc_datetime(d.content["date"]) > latestDate

return lambda data: list(filter(_is_latest, data))
```

Then you can extend the `_is_latest` function in a similar way as above.